packages/toolkit/src/view/model/ModelsList.tsx

```diff
@@ -0,0 +1,60 @@
+"use client";
+
+import {
+  InfiniteData,
+  QueryObserverResult,
+  RefetchOptions,
+} from "@tanstack/react-query";
+import { CardModel } from "../../components/card-model/CardModel";
+import {
+  InstillStore,
+  ListUserModelsResponse,
+  Model,
+  Nullable,
+  useInstillStore,
+  useShallow,
+} from "../../lib";
+import { CardModelSkeleton } from "../../components/card-model/Skeleton";
+
+export type ModelsListProps = {
+  models: Model[];
+  onModelDelete: (
+    options?: RefetchOptions | undefined
+  ) => Promise<
+    QueryObserverResult<InfiniteData<ListUserModelsResponse, unknown>, Error>
+  >;
+  isLoading: boolean;
+};
+
+const selector = (store: InstillStore) => ({
+  accessToken: store.accessToken,
+  enabledQuery: store.enabledQuery,
```

Comment:
@orangecoloured If we don't need the state here, we need to remove it from the selector